tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+   Copyright 2016 Statoil ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include <opm-common/tests/test_CompareEclipseRestart.hpp>
+
+//------------------------------------------------//
+
+int main(int argc, char** argv) {
+    if (argc != 5) {
+        printHelp();
+    }
+    else {
+        try {
+            const char* gridFile1 = argv[1];
+            const char* gridFile2 = argv[2];
+            const char* unrstFile1 = argv[3];
+            const char* unrstFile2 = argv[4];
+            // Comparing grid sizes from .EGRID-files
+            {
+                ecl_grid_type* ecl_grid1 = ecl_grid_alloc(gridFile1);
+                ecl_grid_type* ecl_grid2 = ecl_grid_alloc(gridFile2); 
+                std::cout << "\nName of grid1: " << ecl_grid_get_name(ecl_grid1) << std::endl;
+                const int gridCount1 = ecl_grid_get_global_size(ecl_grid1);
+                std::cout << "Grid1 count = " << gridCount1 << std::endl;
+                std::cout << "Name of grid2: " << ecl_grid_get_name(ecl_grid2) << std::endl;
+                const int gridCount2 = ecl_grid_get_global_size(ecl_grid2);
+                std::cout << "Grid2 count = " << gridCount2 << std::endl;
+
+                ecl_grid_free(ecl_grid1);
+                ecl_grid_free(ecl_grid2);
+            }
+
+            //Comparing keyword values from .UNRST-files:
+
+            ReadUNRST read;
```

Comment:
Naming classes and variables as verbs is unusual. It is often better to have method names be verbs. Such as class UNRSTReader for example. Also, the open() and close() methods should be made into constructor and destructor instead. The bool return from open() is moot anyway, since you (correctly I think) throw if anything goes wrong.